DatadogCore/Sources/Datadog.swift

```diff
@@ -237,37 +242,70 @@ public struct Datadog {
 
     /// Initializes the Datadog SDK.
     ///
+    /// You **must** initialize the core instance of the Datadog SDK prior to enabling any Product.
+    ///
+    ///     Datadog.initialize(
+    ///         with: Datadog.Configuration(clientToken: "<client token>", env: "<environment>"),
+    ///         trackingConsent: .pending
+    ///     )
+    ///
+    /// Once Datadog SDK is initialized, you can enable products, such as RUM:
+    ///
+    ///     RUM.enable(
+    ///         with: RUM.Configuration(applicationID: "<application>")
+    ///     )
+    ///
+    /// It is possible to initialize multiple instances of the SDK, associating them with a name.
+    /// Many methods of the SDK can optionally take a SDK instance as an argument. If not provided,
+    /// the call will be associated with the default (nameless) SDK instance.
+    ///
+    /// To use a secondary instance of the SDK, provide a name to the ``initialize`` method
+    /// and use the returned instance to enable products:
+    ///
+    ///     let core = Datadog.initialize(
+    ///         with: Datadog.Configuration(clientToken: "<client token>", env: "<environment>"),
+    ///         trackingConsent: .pending,
+    ///         instanceName: "my-instance"
+    ///     )
+    ///
+    ///     RUM.enable(
+    ///         with: RUM.Configuration(applicationID: "<application>"),
+    ///         in: core
+    ///     )
```

Comment:
```suggestion
    ///    ```
    ///     Datadog.initialize(
    ///         with: Datadog.Configuration(clientToken: "<client token>", env: "<environment>"),
    ///         trackingConsent: .pending
    ///     )
    ///    ```
    ///
    /// Once Datadog SDK is initialized, you can enable products, such as RUM:
    ///    ```
    ///     RUM.enable(
    ///         with: RUM.Configuration(applicationID: "<application>")
    ///     )
    ///    ```
    /// It is possible to initialize multiple instances of the SDK, associating them with a name.
    /// Many methods of the SDK can optionally take a SDK instance as an argument. If not provided,
    /// the call will be associated with the default (nameless) SDK instance.
    ///
    /// To use a secondary instance of the SDK, provide a name to the ``initialize`` method
    /// and use the returned instance to enable products:
    ///
    ///    ```
    ///     let core = Datadog.initialize(
    ///         with: Datadog.Configuration(clientToken: "<client token>", env: "<environment>"),
    ///         trackingConsent: .pending,
    ///         instanceName: "my-instance"
    ///     )
    ///
    ///     RUM.enable(
    ///         with: RUM.Configuration(applicationID: "<application>"),
    ///         in: core
    ///     )
    ///    ```
```

Reply:
markdown syntax will render nice in our public API docs.

Reply:
https://developer.apple.com/documentation/xcode/formatting-your-documentation-content#Add-code-listings